src/main/scala/com/gradle/internal/Utils.scala

```diff
@@ -47,10 +47,24 @@ object Utils {
     Try(new URI(scheme, host, path, null)).toOption
   }
 
-  private[gradle] def redactUserInfo(url: String): String = {
-    Try(new URI(url)).toOption.flatMap(uri => Option(uri.getUserInfo)) match {
-      case None       => url
-      case Some(info) => url.replace(info + '@', "******@")
+  private[gradle] def redactUserInfo(url: String): Option[String] = {
+    if (!url.startsWith("http")) {
+      return Some(url)
+    }
+
+    try {
+      val uri = new URI(url)
+      val redactedUri = new URI(
+        uri.getScheme(),
+        if (uri.getUserInfo() == null || uri.getUserInfo().isEmpty()) null else "******",
+        uri.getHost(),
+        uri.getPort(),
+        uri.getRawPath(),
+        uri.getRawQuery(),
+        uri.getRawFragment())
+      Some(redactedUri.toString)
+    } catch {
+      case _: URISyntaxException => None
     }
   }
```

Comment:
```suggestion
  private[gradle] def redactUserInfo(url: String): Option[String] =
    if (!url.startsWith("http")) Some(url) else Try(new URI(url)).toOption.map(redactUserInfo).map(_.toString)

  private def redactUserInfo(u: URI): URI =
    new URI(u.getScheme, redact(u.getUserInfo), u.getHost, u.getPort, u.getRawPath, u.getRawQuery, u.getRawFragment)

  private def redact(s: String) = if (s == null || s.isEmpty) null else "******"
```
I made this suggestion with the idea that you wanted to make sure that the user info in the `git` URI wasn't obfuscated, but I'm not really sure that's what we want. Anyway, if you adopt this, make sure to adjust the imports.

Reply:
Another thing to have in mind is that we try to have the different CCUD impl as similar as possible to ease the maintenance. See the [Gradle PR](https://github.com/gradle/common-custom-user-data-gradle-plugin/pull/396/files) for example

Reply:
I'm referring to the test where you validate the the git username is preserved.

Reply:
Applied the suggested change, thanks @stefanobaghino